### PR TITLE
Fixed hidden bugs

### LINKED ISSUE #600: Hidden bugs
It looks something in test suite is breaking PHPUnit error handler. Probably `AutoloadSourceLocator`.

1.

```
vendor/bin/phpunit test\unit\Reflection\ReflectionClassTest.php --filter=testAddMethod

There was 1 error:

1) Roave\BetterReflectionTest\Reflection\ReflectionClassTest::testAddMethod
Undefined property: Roave\BetterReflection\Reflection\ReflectionClass::$cachedMethods
```

2. 

```
vendor/bin/phpunit test\unit\Reflection\StringCast\ReflectionClassConstantStringCastTest.php

There was 1 error:

1) Roave\BetterReflectionTest\Reflection\StringCast\ReflectionClassConstantStringCastTest::testToString with data set #3 ('NO_VISIBILITY_CONSTANT', 'Constant [ public array NO_VI...ay }\n')
assert(): assert($value === null || is_scalar($value)) failed
```

3. 

```
vendor/bin/phpunit test\unit\Reflection\ReflectionObjectTest.php

There were 20 errors:

1) Roave\BetterReflectionTest\Reflection\ReflectionObjectTest::testReflectionObjectOverridesAllMethodsInReflectionClass with data set "getMethods" ('getMethods')
Function ReflectionType::__toString() is deprecated
```

Probably even more.

### PR DESCRIPTION
**Pull request by kukulich**
Fixes #600

### CODE REVIEW

File: src/Reflection/StringCast/ReflectionClassConstantStringCast.php [hunk 1]
```diff
@@ -18,14 +17,13 @@ final class ReflectionClassConstantStringCast
...
         $value = $constantReflection->getValue();
-        assert($value === null || is_scalar($value));
 
         return sprintf(
             "Constant [ %s %s %s ] { %s }\n",
             self::visibilityToString($constantReflection),
             gettype($value),
             $constantReflection->getName(),
-            (string) $value
+            is_array($value) ? 'Array' : (string) $value
```
Comment (Ocramius) [inline src/Reflection/StringCast/ReflectionClassConstantStringCast.php:26]: Mentioned elsewhere: we need a test verifying https://3v4l.org/q8nvd

Comment (kukulich) [reply src/Reflection/StringCast/ReflectionClassConstantStringCast.php:26]: @Ocramius The test exists already: https://github.com/Roave/BetterReflection/blob/master/test/unit/Reflection/StringCast/ReflectionClassConstantStringCastTest.php#L35

Comment (Ocramius) [reply src/Reflection/StringCast/ReflectionClassConstantStringCast.php:26]: Ah, interesting! Thanks for hunting it down: guess the error handler was not being tripped :-)

**Comment by Ocramius** [comment]
Thanks again: I'll rebase my work locally to see if I can notice more errors :+1:

**Pull request merged**